coursier.bzl

```diff
@@ -261,6 +261,50 @@ def get_home_netrc_contents(repository_ctx):
                 return repository_ctx.read(netrcfile)
     return ""
 
+def generate_coursier_credentials_from_netrc(netrc_content):
+    """Generate a list of coursier credentials from a netrc file. Only
+    'machine' entries containing a login and a password are taken into
+    account. The 'default' entry is ignored.
+    """
+    spacified = netrc_content.replace("\n", " ").replace("\t", " ")
+    sanitized = [e for e in spacified.split(" ") if e != ""]
```

Comment:
Would be good to ignore `#` style comments. The two netrc parsers in Bazel both do this:

https://github.com/bazelbuild/bazel/blob/4c0febc02fab5e9fda3f40532a62854fb1c45631/tools/build_defs/repo/utils.bzl#L228

https://github.com/bazelbuild/bazel/pull/12130

Reply:
Also the second one checks the `$NETRC` env var as an alternative to the hardcoded `~/.netrc`, would be nice to support that too.

Reply:
I changed the function to use the read_netrc helper provided by Bazel.